outlook/src/classes/Company.ts

```diff
@@ -33,10 +42,27 @@ class Company {
         this.additionalInfo = {};
     }
 
-    static fromJSON(o: Object): Company {
+    static fromJSON(o: Object, enrichmentInfo?: EnrichmentInfo): Company {
         if (!o) return new Company();
         const company = Object.assign(new Company(), o);
         company.address = Address.fromJSON(o['address']);
+
+        if (enrichmentInfo != null) {
+            if (enrichmentInfo.type == EnrichmentInfoType.NoData) {
+                company.enrichmentStatus = EnrichmentStatus.enrichmentEmpty;
+            } else {
+                if (EnrichmentInfoType.CompanyCreated || EnrichmentInfoType.CompanyUpdated) {
+                    company.enrichmentStatus = EnrichmentStatus.enriched;
+                }
+            }
+        } else {
```

Comment:
elif - else ? That way we avoid too much nested if constructs.

Reply:
No elif in TypeScript unfortunately :/

Reply:
"else if" should work ^^

Reply:
Oh yes, done, we are now using if and else if.